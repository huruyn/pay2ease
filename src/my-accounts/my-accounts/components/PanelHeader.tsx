import PropTypes from 'prop-types';

function PanelHeader({
  title,
  sub_title,
  border_color,
}: {
  title: string;
  sub_title: string;
  border_color: string;
}) {
  return (
    <div
      className="d-flex align-items-center"
      style={{
        borderTop: `3px solid ${border_color}`,
        borderBottom: '2px solid grey',
        paddingTop: 10,
        paddingBottom: 10,
        backgroundColor: '#F2F3F2',
        paddingLeft: 5,
        marginTop: 10,
      }}
    >
      <span style={{ fontWeight: 500 }}>
        {title}
        {' '}
        |
        {' '}
      </span>
      <a
        href="/"
        className="text-decoration-none"
        style={{ marginLeft: 5, fontSize: 14, color: 'rgb(220, 53, 69)' }}
      >
        {sub_title}
      </a>
    </div>
  );
}

PanelHeader.propTypes = {
  title: PropTypes.string.isRequired,
  border_color: PropTypes.string,
  sub_title: PropTypes.string,
};

PanelHeader.defaultProps = {
  border_color: 'green',
  sub_title: '',
};

export default PanelHeader;
