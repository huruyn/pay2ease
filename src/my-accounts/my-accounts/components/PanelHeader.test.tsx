import React from 'react';
import PanelHeader from './PanelHeader';

const ReactTestRenderer = require('react-test-renderer');

describe('GetAlertedHeader Component', () => {
  it('matches the snapshot', () => {
    const tree = ReactTestRenderer.create(
      <PanelHeader title="Test Title" />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
