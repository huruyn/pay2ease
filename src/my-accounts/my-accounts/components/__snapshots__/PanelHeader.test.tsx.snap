// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetAlertedHeader Component matches the snapshot 1`] = `
<div
  className="d-flex align-items-center"
  style={
    Object {
      "backgroundColor": "#F2F3F2",
      "borderBottom": "2px solid grey",
      "borderTop": "3px solid green",
      "marginTop": 10,
      "paddingBottom": 10,
      "paddingLeft": 5,
      "paddingTop": 10,
    }
  }
>
  <span
    style={
      Object {
        "fontWeight": 500,
      }
    }
  >
    Test Title
     
    |
     
  </span>
  <a
    className="text-decoration-none"
    href="/"
    style={
      Object {
        "color": "rgb(220, 53, 69)",
        "fontSize": 14,
        "marginLeft": 5,
      }
    }
  >
    
  </a>
</div>
`;
